### PR TITLE
Adds a small cafeteria behind the right wing shutters of the museum.

### PR DESCRIPTION
**Pull request by Ghommie**
## About The Pull Request
I was thinking to contribute something to the new away mission map to make it better. Mapping and all takes too much time for me, so I could do little. Though it comes with its own unique gimmicks.

To reach the cafeteria, one has to complete a couple puzzles.
The first set is opened by inputing the correct PIN on the password panel beside it. There're several clues to help you guess this fairly easy puzzle, in the form of several number graffitis, a scrapped piece of paper full of numbers, and a board filled with colored dots also found just beside the panel.
The second one is opened by a keycard, and is generally lazier. To find it, you'll need to do a bit of (toilet) searching.

As for the unique things this PR adds:
- A fire extinguisher... that actually contains welding fuel
- A (dirt-cheap) hotdog vending machine*
- A completely ornamental maneki-neko (that's the name of the luck-bringing, paw-waving cat figurine)
- A piggy bank that carries money between rounds. It has a cap of 10k credits worth of holochips, cash and coins, which is pretty high, but I'm confident people will just destroy it for its contents the moment they find it. His name is Pigston Swinelord VI.
- More, totally legit and not actually fake bombable walls :^)

*By the by, you can also find it during the national hotdog day.

Screenshots of the new location:
museum cafe
museum restrooms

## Why It's Good For The Game
You know how most away missions are not that special at all? Yeah, @mc-oofert set an example of a pretty decent one actually, if not a tad small. I thought it could use a touch of another mind actually contributing to it too, because it deserves it.

Also, this sets the basis for other persistent piggy banks. I don't think they should all have that 10k cap like this one, perhaps 1k is enough. Beside, the code that mothblocks did for json database datum is pretty good, so there is not a whole lot of shitcode here.

## Changelog

:cl:
add: Added a cafeteria to the museum away mission, with a few special things to it. To reach it, you'll have to complete a couple puzzles however.
map: The museum away mission now has a couple restrooms.
add: Hotdog vending machines may spawn during the National Hot Dog Day.
/:cl:

### CODE REVIEW

**Comment by Ghommie** [comment]
Drafting, because of course it needs pictures.

**Comment by mc-oofert** [comment]
i intended the shutters north to be much more of a silly secret found for stealing the keycard from a mothroach which is also pretty hidden so i think it would be probably just better if there was some other requirement or just not that specific keycard

**Comment by Loafin34** [comment]
you should make Mayonnaise's description into "not an instrument."

**Comment by Ghommie** [comment]
> i intended the shutters north to be much more of a silly secret found for stealing the keycard from a mothroach which is also pretty hidden so i think it would be probably just better if there was some other requirement or just not that specific keycard

Sure, I'll think something about it.

**Comment by Ghommie** [comment]
Ok, I've made a new puzzle for the first set of shutters.

**Comment by Wallemations** [comment]
you said you were drafting this for that, but this tag is just to remind me/others at a glance

File: code/game/objects/items/busts_and_figurines.dm [hunk 1]
```diff
@@ -0,0 +1,138 @@
...
+	w_class = WEIGHT_CLASS_SMALL
+	force = 5
+	throwforce = 5
+	throw_speed = 3
+	throw_range = 5
+	attack_verb_continuous = list("bashes", "beckons", "hit")
+	attack_verb_simple = list("bash", "beckon", "hit")
+
+/obj/item/maneki_neko/Initialize(mapload)
+	color = pick(3;COLOR_WHITE, 2;COLOR_GOLD, 1;COLOR_DARK) //Not compatible with greyscale configs because it's animated.
```
Comment (JohnFulpWillard) [inline code/game/objects/items/busts_and_figurines.dm:133]: Shouldn't this use pick weighted or something?

File: code/game/objects/items/piggy_bank.dm [hunk 2]
```diff
@@ -0,0 +1,126 @@
...
+
+/obj/item/piggy_bank/attackby(obj/item/item, mob/user, params)
+	var/creds_value = item.get_item_credit_value()
+	if(isnull(creds_value))
+		return ..()
+
+	if(calculate_dosh_amount() + creds_value > maximum_value)
+		balloon_alert(user, "cannot hold so much cash!")
+	else if(!user.transferItemToLoc(item, src))
+		balloon_alert(user, "item stuck in your hands!")
```
Comment (JohnFulpWillard) [inline code/game/objects/items/piggy_bank.dm:104]: These can probably be shorter

File: code/game/objects/items/piggy_bank.dm [hunk 3]
```diff
@@ -0,0 +1,126 @@
...
+	else
+		balloon_alert(user, "inserted [creds_value] credits")
+	return TRUE
+
+///Returns the total amount of credits that its contents amount to.
+/obj/item/piggy_bank/proc/calculate_dosh_amount()
+	var/total_value = 0
+	for(var/obj/item/item in src)
```
Comment (JohnFulpWillard) [inline code/game/objects/items/piggy_bank.dm:112]: Why not check contents instead?

File: code/game/objects/items/puzzle_pieces.dm [hunk 4]
```diff
@@ -363,13 +362,75 @@ MAPPING_DIRECTIONAL_HELPERS(/obj/machinery/puzzle_button, 32)
...
+	///associate a color to each digit that may be found in the password.
+	var/list/digit_to_color = list()
+
+/obj/machinery/puzzle/password/pin/Initialize(mapload)
+	. = ..()
+
+	for(var/iteration in 1 to pin_length)
+		password += "[rand(1, 9)]"
+
+	var/possible_colors = list(
```
Comment (JohnFulpWillard) [inline code/game/objects/items/puzzle_pieces.dm:419]:
```suggestion
	var/list/possible_colors = list(
```

File: code/game/objects/items/puzzle_pieces.dm [hunk 5]
```diff
@@ -461,3 +522,99 @@ MAPPING_DIRECTIONAL_HELPERS(/obj/machinery/puzzle_keycardpad, 32)
...
+		qdel(src)
+		return FALSE
+	icon_state = pick(pass_character)
+	if(!text2num(icon_state))
+		name = "letter"
+		desc = "A letter vandalizing the station."
+	return TRUE
+
+/obj/effect/decal/cleanable/crayon/puzzle/pin
+	name = "graffiti that's clue to a PIN puzzle"
```
Comment (JohnFulpWillard) [inline code/game/objects/items/puzzle_pieces.dm:596]: this name feels really odd to read, can it be something like "a potential clue" or "PIN number"?

File: tgstation.dmm-pal [hunk 6]
```diff
@@ -0,0 +1 @@
+[Palette]
```
Comment (JohnFulpWillard) [inline tgstation.dmm-pal:1]: What's this file for?

Comment (Ghommie) [reply tgstation.dmm-pal:1]: Oh, it seems using the built-in map editor generates this file now. I should make a PR to include it in git-ignore.

**Review by Maurukas — CHANGES_REQUESTED** [review]
image

One issue. Top door has the id_tag var set instead of id so it does not unlock.

**Review by Maurukas — APPROVED** [review]

**Pull request merged**